public/comments.php

```diff
@@ -22,135 +28,121 @@
+        $pageTitle = renderGameTitle($game->Title . ' (' . $console->Name . ')');
         $commentsLabel = "Hash Comments";
-        $navPath =
-        [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
-            'Manage Hashes' => '/managehashes.php?g=' . $gameData['ID'],
+        $navPath = [
+            '_GamePrefix' => renderGameBreadcrumb($game->ID),
+            'Manage Hashes' => '/managehashes.php?g=' . $game->ID,
         ];
         break;
 
     case ArticleType::GameModification:
         if ($permissions < Permissions::JuniorDeveloper) {
             abort(403);
         }
-        $gameData = getGameData($articleID);
-        if ($gameData === null) {
-            abort(404);
-        }
-        $pageTitle = renderGameTitle($gameData['Title'] . ' (' . $gameData['ConsoleName'] . ')');
+        /** @var Game $game */
+        $game = Game::findOrFail($articleID);
+        /** @var System $console */
+        $console = $game->console;
+        $pageTitle = renderGameTitle($game->Title . ' (' . $console->Name . ')');
         $commentsLabel = "Modifications";
         $navPath =
         [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
+            '_GamePrefix' => renderGameBreadcrumb($game->ID),
         ];
         break;
 
     case ArticleType::SetClaim:
         if ($permissions < Permissions::Admin) {
             abort(403);
         }
-        $gameData = getGameData($articleID);
-        if ($gameData === null) {
-            abort(404);
-        }
-        $pageTitle = renderGameTitle($gameData['Title'] . ' (' . $gameData['ConsoleName'] . ')');
+        /** @var Game $game */
+        $game = Game::findOrFail($articleID);
+        /** @var System $console */
+        $console = $game->console;
+        $pageTitle = renderGameTitle($game->Title . ' (' . $console->Name . ')');
         $commentsLabel = "Claim Comments";
-        $navPath =
-        [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
-            'Manage Claims' => '/manageclaims.php?g=' . $gameData['ID'],
+        $navPath = [
+            '_GamePrefix' => renderGameBreadcrumb($game->ID),
+            'Manage Claims' => '/manageclaims.php?g=' . $game->ID,
         ];
         break;
 
     case ArticleType::Achievement:
-        $pageTitle = getAchievementTitle($articleID, $gameTitle, $gameID);
-        if (empty($pageTitle)) {
-            abort(404);
-        }
-        $gameData = getGameData($gameID);
-        if ($gameData === null) {
-            abort(404);
-        }
-        $navPath =
-        [
-            '_GamePrefix' => renderGameBreadcrumb($gameData),
-            $pageTitle => '/achievement/' . $articleID,
+        /** @var Achievement $achievement */
+        $achievement = Achievement::findOrFail($articleID);
+        /** @var Game $game */
+        $game = Game::findOrFail($achievement->GameID);
+        /** @var System $console */
+        $console = $game->console;
```

Comment:
$console is unused

Reply:
Ah yes - those are remnants from before `renderGameBreadcrumb()` was introduced. Or i copied too much? Anyways, removed.